packages/nc-gui/pages/[projectType]/[projectId]/index/index.vue

```diff
@@ -26,6 +28,16 @@ const { isOpen, toggle } = useSidebar('nc-left-sidebar')
 function onEdit(targetKey: number, action: 'add' | 'remove' | string) {
   if (action === 'remove') closeTab(targetKey)
 }
+
+const hideSidebarOnClickOrTouchIfMobileMode = () => {
+  if (isMobileMode.value && isOpen.value) {
+    toggle(false)
+  }
+}
+
+onMounted(() => {
+  mainArea.value?.addEventListener('click', hideSidebarOnClickOrTouchIfMobileMode)
+})
```

Comment:
Any reason to not use `@click` at the div to listen click event.

```
<div ref="mainArea" class="w-full min-h-[300px] flex-auto" @click="hideSidebarOnClickOrTouchIfMobileMode">
```

Reply:
Very good question 😂 
I changed it as suggested